### DISCUSSION TITLE
Are all dynamic variables available to Flows operations?

### DISCUSSION THREAD

**Discussion by urbgimtam** [directus/directus#14222] [Archive]
I've been trying to create a flow to delete items of a collection when the current time is larger than a "datetime" field in the required collection.

The Flow should be really basic:

- A "Cron" trigger (running every 10 minutes)
- A "Delete" operation with a query where:
- - filters "datetime" field is less than the dynamic variable $NOW.

However, the operations' console returns the query with "undefined" when the value $NOW (the server's datetime) should be.
The documentation says the query uses the Filter Rules. 
The Filter Rules documentation says you can use $NOW.

The query I'm testing is this:
```
{
  "filter": {
    "dateToArchive": {
      "_lte": "{{ $NOW() }}"
    }
  }
}
```

This is the "option" log of the operation. 
```
{
  "collection": "finalComposition_components",
  "query": "{
                    "filter": {
                        "dateToArchive": {
                            "_lte": "undefined"
                         }
                     }
                  }",
  "permissions": "$full",
  "key": []
}
```

As you can see, the $NOW becomes "undefined".

Could it be a bug, or is it not supported? (if so, it should be explained in the docs)

**Comment by rijkvanzanten** [comment]
The query would be

```json
{
  "filter": {
    "dateToArchive": {
      "_lte": "$NOW"
    }
  }
}
```

instead of 

```json
{
  "filter": {
    "dateToArchive": {
      "_lte": "{{ $NOW() }}"
    }
  }
}
```

🙂 

The `{{}}` pair indicates that you want to use data from another flow, so in this example it's trying to read the data from the previous flow operation with the key `$NOW`, which doesn't exist 👍🏻

**Reply by urbgimtam** [reply]
Thank you for the clarification.

I did try that first, but was confused by the Operations' option log result, as the $NOW variable doesn't resolve on the Options Log

When using 
```
{
  "filter": {
    "dateToArchive": {
      "_lte": "$NOW"
    }
  }
}
```

the log returns 
```
{
  "collection": "finalComposition_components",
      "query": "{
           "filter": {
                "dateToArchive": {
                    "_lte": "$NOW"
                 }
         }
   }",
  "permissions": "$full",
  "key": []
}
```

but with `{{$NOW}}` it would return `undefined`.

This led me to think the later was the correct syntax. It was one of my first attempts to make a Flow, and using dynamic variables.

Thank you again for the clarification.

**Reply by rijkvanzanten** [reply]
No problem. I can see how that's a little confusing. In this case, it's the API that converts the $NOW to the current timestamp the moment the filter is used, rather than the Flows setting up the options ahead of time
